core/psh/pshapp/auth.c

```diff
@@ -139,12 +139,12 @@ static int psh_login(int argc, char **argv)
 		return 1;
 
 	sleep(2);
-	return -1;
+	return 0;
 }
 
 
-void __attribute__((constructor)) login_registerapp(void)
+void __attribute__((constructor)) pshauth_registerapp(void)
 {
-	static psh_appentry_t app = {.name = "login", .run = psh_login, .info = NULL};
+	static psh_appentry_t app = { .name = "auth", .run = psh_auth, .info = NULL };
```

Comment:
please note that `auth` symlink will not be created, I don't know if it's intentional

Reply:
Yes, that is intentional. `auth` is just a command registered by `pshapp` for internal use - not to be called under that name. There are currently no application that would just call the `psh` shell to authenticate user. Even if such application will be created it will be just a matter of adding `auth` to `PSH_LINK_APPLETS`
 
As for now the `auth` applet is only used in `pshapp` so it may also be just a function call, but the future goal is to allow other `psh` applets to call `auth` for user authentication when user privileges will be a thing (e.g `edit` opening a file).